src/sentry/integrations/slack/utils/channel.py

```diff
@@ -188,26 +198,64 @@ def get_channel_id_with_timeout(
     return prefix, channel_id, False
 
 
-def check_for_channel(client: SlackClient, name: str) -> str | None:
-    msg_response = client.post(
-        "/chat.scheduleMessage",
-        data={
-            "channel": name,
-            "text": "Sentry is verifying your channel is accessible for sending you alert rule notifications",
-            "post_at": int(time.time() + 500),
-        },
-    )
-
-    if "channel" in msg_response:
-        client.post(
-            "/chat.deleteScheduledMessage",
-            params=dict(
-                {
-                    "channel": msg_response["channel"],
+def check_for_channel(
+    client: SlackClient | SlackSdkClient, name: str, has_sdk_flag: bool
+) -> str | None:
+    if has_sdk_flag:
+        try:
+            msg_response = client.chat_scheduleMessage(
+                channel=name,
+                text="Sentry is verifying your channel is accessible for sending you alert rule notifications",
+                post_at=int(time.time() + 500),
+            )
+        except SlackApiError as e:
+            logger_params = {
+                "error": str(e),
+                "integration_id": client.integration_id,
+                "channel": name,
+                "post_at": int(time.time() + 500),
+            }
+            logger.exception("slack.chat_scheduleMessage.error", extra=logger_params)
+            raise
+
+        if "channel" in msg_response:
```

Comment:
what should we do if channel is not in msg_response? This would simply return `None` right?

Reply:
could we just early return here?
```
if "channel" not in msg_response:
    return None
```